src/async-form/reducer.js

```diff
@@ -0,0 +1,83 @@
+import { forKeys } from '../utils/object';
+import { createReducer as createDefaultReducer } from '../async-data/reducer';
+import merge from 'merge';
+
+export const addDataToState = (state, data) => merge.recursive(true, state, { data });
+
+export const selectData = ({ data }) => data;
+
+const separator = '.';
+
+const updateDataForName = (data, path, value) => {
+	if (path.length === 1) {
+		data[path[0]] = value;
+	} else {
+		updateDataForName(data[path[0]], path.slice(1), value);
+	}
+};
+
+export const updateData = ({ data }, { name, value }) => {
+	data = JSON.parse(JSON.stringify(data));
```

Comment:
Maybe for copying this it's cleaner to do it with merge considering you already have it imported

Reply:
👍